keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -111,38 +111,35 @@ def _flip_image(image, transformation):
         return flipped_output
 
     def _flip_bounding_boxes_horizontal(bounding_boxes):
-        x1, x2, x3, x4, rest = tf.split(
-            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
-        )
+        x1, x2, x3, x4 = tf.split(bounding_boxes["boxes"], [1, 1, 1, 1], axis=-1)
         output = tf.stack(
             [
                 1 - x3,
                 x2,
                 1 - x1,
                 x4,
-                rest,
             ],
             axis=-1,
         )
-        output = tf.squeeze(output, axis=1)
-        return output
+        bounding_boxes = bounding_boxes.copy()
+        bounding_boxes["boxes"] = tf.squeeze(output, axis=1)
```

Comment:
If user run this eagerly, this will actually copy the tensor, I think?

Reply:
the tf.squeeze() or bounding_boxes.copy()?  

In any case, do we care at all about eager performance?  From my understanding keras does not care about eager performance as it is already really bad.

Reply:
Ok, we could revisit this if memory issue comes up